### PR TITLE
fix(printing): Fix print button on ContentPreview

### PR DESCRIPTION
**Pull request by mickr**
* Print button should not show for non-printable file-types

### CODE REVIEW

**Comment by mickr** [comment]
Dependent upon https://github.com/box/box-content-preview/pull/1177

File: src/elements/content-preview/ContentPreview.js [hunk 1]
```diff
@@ -686,6 +690,26 @@ class ContentPreview extends React.PureComponent<Props, State> {
...
+     * Returns whether a preview viewer supports printing
+     *
+     * @return {boolean}
+     */
+    handleCanPrint() {
+        /*
+        Setting this as default so older preview versions can still print documents without having the
+        default state have the print icon.
+         */
+        let canPrint = true;
```
Comment (ConradJChan) [inline src/elements/content-preview/ContentPreview.js]: Would it make sense to default `canPrint` state to `true` then?

Comment (mickr) [reply src/elements/content-preview/ContentPreview.js]: Talked about this off-line, setting it to true will flash the icon and then remove it for files that cannot print.

File: src/elements/content-preview/ContentPreview.js [hunk 2]
```diff
@@ -686,6 +690,26 @@ class ContentPreview extends React.PureComponent<Props, State> {
...
+        const preview = this.getPreview();
+
+        if (preview && typeof preview.canPrint === 'function') {
+            canPrint = preview.canPrint();
+        }
+
+        return this.setState({ canPrint });
```
Comment (jstoffan) [inline src/elements/content-preview/ContentPreview.js]: Can be simplified and `return` can be removed.

```
const preview = this.getPreview();

this.setState({
    canPrint: preview && (!preview.canPrint || preview.canPrint())
});
```

File: src/elements/content-preview/ContentPreview.js [hunk 3]
```diff
@@ -686,6 +690,26 @@ class ContentPreview extends React.PureComponent<Props, State> {
         return !!showAnnotations && (this.canAnnotate() || hasViewAllPermissions || hasViewSelfPermissions);
     }
 
+    /**
+     * Returns whether a preview viewer supports printing
+     *
+     * @return {boolean}
```
Comment (jstoffan) [inline src/elements/content-preview/ContentPreview.js]: Accurate? We generally leave JSDocs off of new code due to their tendency to become outdated.

File: src/elements/content-preview/ContentPreview.js [hunk 4]
```diff
@@ -686,6 +690,26 @@ class ContentPreview extends React.PureComponent<Props, State> {
...
+     */
+    handleCanPrint() {
+        /*
+        Setting this as default so older preview versions can still print documents without having the
+        default state have the print icon.
+         */
+        let canPrint = true;
+        const preview = this.getPreview();
+
+        if (preview && typeof preview.canPrint === 'function') {
```
Comment (jstoffan) [inline src/elements/content-preview/ContentPreview.js]: This method is only called in `onPreviewLoad`. Is it possible for preview to be falsy?

File: src/elements/content-preview/stories/ContentPreview.stories.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+// @flow
```
Comment (jstoffan) [inline src/elements/content-preview/stories/ContentPreview.stories.js]: Should these stories be introduced in a separate PR?

File: src/elements/content-preview/ContentPreview.js (hunk 4 above)
Comment (mickr) [reply src/elements/content-preview/ContentPreview.js]: Not in this scenario because the preview is loaded. But if this gets used outside of that scope it could because this.getPreview will return null if there isn't a preview instance or file, so I felt it better to be safe.

File: src/elements/content-preview/ContentPreview.js [hunk 6]
```diff
@@ -686,6 +690,11 @@ class ContentPreview extends React.PureComponent<Props, State> {
         return !!showAnnotations && (this.canAnnotate() || hasViewAllPermissions || hasViewSelfPermissions);
     }
 
+    handleCanPrint() {
+        const preview = this.getPreview();
+        this.setState({ canPrint: !!preview && preview.canPrint && preview.canPrint() });
```
Comment (mxiao6) [inline src/elements/content-preview/ContentPreview.js]: why the `preview.canPrint` is necessary

Comment (mxiao6) [reply src/elements/content-preview/ContentPreview.js]: Should it be `!!preview.canPrint` then?
